dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -245,21 +245,32 @@ public async Task<Plan> InvokeNextStepAsync(SKContext context, CancellationToken
         if (this.HasNextStep)
         {
             var step = this.Steps[this.NextStepIndex];
-
             // Merge the state with the current context variables for step execution
             var functionVariables = this.GetNextStepVariables(context.Variables, step);
 
             // Execute the step
             var functionContext = new SKContext(functionVariables, context.Skills, context.Logger);
+
+            string? renderedPrompt = null;
+            if (step.Function is not null && step.Function is SemanticFunction semanticFunction)
```

Comment:
A few things feel odd:

- Calling RenderAsync twice for a single semantic function
- Every step in a plan will call the handler, but if I have a semantic or native function that's composed of native and semantic functions, I'll get only a single callback.